pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -41,6 +41,33 @@ func metricsTimerTask(ctx *domainContext, hyper hypervisor.Hypervisor) {
 	}
 }
 
+func logWatermarks(ctx *domainContext, status *types.DomainStatus, dm *types.DomainMetric) {
+	if status == nil {
+		return
+	}
+
+	config := lookupDomainConfig(ctx, status.Key())
+	if config == nil {
+		return
+	}
+
+	var CurrMaxUsedMemory uint32
+	st, _ := ctx.pubDomainMetric.Get(dm.Key())
+	if st != nil {
+		previousMetric := st.(types.DomainMetric)
+		CurrMaxUsedMemory = previousMetric.MaxUsedMemory
+	}
+
+	if CurrMaxUsedMemory < dm.MaxUsedMemory {
+		usedPercents := dm.MaxUsedMemory * 100 * 1024 / uint32(config.Memory)
```

Comment:
I don't see any checks in zedmanager, domainmgr, or the hypervisor package where we outright reject an app instance with config.Memory = 0. (It would fail to boot I assume.)
So guarding against divide by zero makes sense.

Reply:
right. I was assuming that qemu would not start with 0 memory. And in case of container writing 0 to `memory.limit_in_bytes` would mean that application is not allowed to have any memory and will be killed right away.

> It would fail to boot I assume
isn't this config set before we try too boot? 

But you are right, it is worth guarding, since panic in Pillar must be avoided even if this should never happen.